src/service/impl/stop-places/index.ts

```diff
@@ -0,0 +1,107 @@
+import {IStopPlacesService} from '../../interface';
+import {journeyPlannerClient} from '../../../graphql/graphql-client';
+import {Result} from '@badrap/result';
+import {
+  GetStopPlaceConnectionsDocument,
+  GetStopPlaceConnectionsQuery,
+  GetStopPlaceConnectionsQueryVariables,
+} from './journey-gql/stop-place-connections.graphql-gen';
+import {isDefined, onlyUniquesBasedOnField} from './utils';
+import {APIError} from '../../../utils/api-error';
+import {
+  GetStopPlacesByModeDocument,
+  GetStopPlacesByModeQuery,
+  GetStopPlacesByModeQueryVariables,
+} from './journey-gql/stop-places-mode.graphql-gen';
+import {QuayFragment} from '../fragments/journey-gql/quays.graphql-gen';
+import {JourneyPatternsFragment} from '../fragments/journey-gql/journey-pattern.graphql-gen';
+
+export default (): IStopPlacesService => {
+  return {
+    async getStopPlacesByMode(query, headers) {
+      const result = await journeyPlannerClient(headers).query<
+        GetStopPlacesByModeQuery,
+        GetStopPlacesByModeQueryVariables
+      >({
+        query: GetStopPlacesByModeDocument,
+        variables: {
+          authorities: query.authorities,
+          transportModes: query.transportModes,
+        },
+      });
+
+      if (result.errors) {
+        return Result.err(new APIError(result.errors));
+      }
+
+      try {
+        const uniqueStopPlaces = result.data.lines
+          .filter((line) =>
+            query.transportSubmodes?.find(
+              (subMode) => subMode === line.transportSubmode,
+            ),
+          )
```

Comment:
This will always filter out all lines if no `transportSubmodes` is sent in the query.